spin/cmds/meson.py

```diff
@@ -124,6 +134,31 @@ def _meson_version_configured():
         pass
 
 
+def _check_coverage_tool_installation(coverage_type: GcovReports):
+    requirements = {  # https://github.com/mesonbuild/meson/blob/6e381714c7cb15877e2bcaa304b93c212252ada3/docs/markdown/Unit-tests.md?plain=1#L49-L62
+        GcovReports.html: ["Gcovr/GenHTML", "lcov"],
+        GcovReports.xml: ["Gcovr (version 3.3 or higher)"],
+        GcovReports.text: ["Gcovr (version 3.3 or higher)"],
+        GcovReports.sonarqube: ["Gcovr (version 4.2 or higher)"],
+    }
+
+    # First check the presence of a valid build
+    if not (os.path.exists(build_dir)):
+        raise click.ClickException(
+            "`build` folder not found, cannot generate coverage reports. "
+            "Generate coverage artefacts by running `spin test --gcov`"
+        )
+
+    # Verify the tools are installed prior to the build
+    p = _run(["ninja", "-C", build_dir, "-t", "targets", "all"], output=False)
+    if f"coverage-{coverage_type.value}" not in p.stdout.decode("ascii"):
+        raise click.ClickException(
+            f"coverage-{coverage_type.value} is not supported... "
+            f"Ensure the following are installed: {', '.join(requirements[coverage_type])} "
```

Comment:
This error is raised when `spin build; spin test --gcov-report`. Needs to ensure that build was run with `--gcov`.

Reply:
Yep, shall I add a dependency to call `spin build --gcov` in case it's missing?

Reply:
Ah ok, I get the problem. We need to check for coverage artifacts. Let me add it.

Reply:
Added checks and UT in f2c21eb9d658e5a22dbd991bd217dd52030f0d12